omnigibson/utils/physx_utils.py

```diff
@@ -226,11 +213,21 @@ def create_physx_particleset_pointinstancer(
     instancer.GetAngularVelocitiesAttr().Set(Vt.Vec3fArray.FromNumpy(angular_velocities))
     instancer.GetScalesAttr().Set(Vt.Vec3fArray.FromNumpy(scales))
 
+    # Take a render step to "lock" the visuals of the prototypes at the graveyard position
+    # This needs to happen AFTER setting particle states
+    og.sim.render()
+
+    # Then we move the prototypes back to zero offset because otherwise all the generated particles will be offset by
+    # the graveyard position. At this point, the prototypes themselves no longer appear at the zero offset (locked at
+    # the graveyard position), which is desirable because we don't want the agent to see the prototypes themselves.
```

Comment:
holy, what is this black magic???

Reply:
This is black magic... I can't think of another way to fix this...